notebooks/225-yolov7-optimization/225-yolov7-optimiization.ipynb

```diff
@@ -0,0 +1,820 @@
+    "It is the reason why we will add only `--grid` parameter to preserve original pytorch model result format.\n",
+    "If you want to understand how to work with end2end ONNX model, you can check this [notebook](https://github.com/WongKinYiu/yolov7/blob/main/tools/YOLOv7onnx.ipynb)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python export.py --weights model/yolov7-tiny.pt --grid"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Convert ONNX Model to OpenVINO Intermideate Representation (IR)\n",
+    "While ONNX models are directly supported by OpenVINO runtime, it can be useful to convert them to IR format to take advantage of OpenVINO optimization tools and features.\n",
+    "`mo.convert` python function can be used for converting model using OpenVINO Model Optimizer. \n",
+    "The function returns instance of OpenVINO Model class, which is ready to use in Python interface but can also be serialized to OpenVINO IR format for future execution."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from openvino.tools import mo\n",
+    "from openvino.runtime import serialize\n",
+    "\n",
+    "model = mo.convert(input_model='model/yolov7-tiny.onnx')\n",
+    "# serialize model for saving IR\n",
+    "serialize(model, 'model/yolov7-tiny.xml')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Verify model inference\n",
+    "\n",
+    "For testing model work, we create inference pipeline similar to `detect.py`. Our pipeline consists from preprocessing step, inference of OpenVINO model and postprocess results for getting bounding boxes."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Preprocessing\n",
+    "\n",
+    "Model input is tensor with shape `[1, 3, 640, 640]` in `N, C, H, W` format, where\n",
+    "* `N` - number of images in batch (batch size)\n",
+    "* `C` - image channels\n",
+    "* `H` - image height\n",
+    "* `W` - image width\n",
+    "\n",
+    "Model expects images in RGB channels format and normalized in [0, 1] range.\n",
+    "For resizing images to fit model size `letterbox` resize approach is used where the aspect ratio of width and height is preserved. It defined in yolov7 repository.\n",
+    "To keep specific shape preprocessing automatically enables padding."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import torch\n",
+    "import cv2\n",
+    "from utils.datasets import letterbox\n",
+    "from utils.plots import plot_one_box\n",
+    "\n",
+    "\n",
+    "def preprocess_image(img0: np.ndarray):\n",
+    "    \"\"\"\n",
+    "    Preprocess image according YOLOv7 input requirements. \n",
```

Comment:
Preprocess image according to YOLOv7 input requirements.

Reply:
fixed, thanks